pkg/scheduler/framework/plugins/legacy_registry.go

```diff
@@ -497,7 +497,7 @@ func NewPluginConfig(pluginName string, args interface{}) config.PluginConfig {
 	}
 	return config.PluginConfig{
 		Name: pluginName,
-		Args: runtime.Unknown{Raw: encoding},
+		Args: runtime.RawExtension{Raw: encoding},
```

Comment:
Q: should we change `args` to another type such as runtime.RawExtension, so that we don't need to marshal (L494) and unmarshal again (I suppose during processing PluginConfig)?

Reply:
A simplified snippet:

```diff
diff --git a/cmd/kube-scheduler/app/options/deprecated.go b/cmd/kube-scheduler/app/options/deprecated.go
index 2becd3c97a2..0cb720883cd 100644
--- a/cmd/kube-scheduler/app/options/deprecated.go
+++ b/cmd/kube-scheduler/app/options/deprecated.go
@@ -129,10 +129,10 @@ func (o *DeprecatedOptions) ApplyTo(cfg *kubeschedulerconfig.KubeSchedulerConfig
 		profile.SchedulerName = o.SchedulerName
 	}
 	if o.HardPodAffinitySymmetricWeight != interpodaffinity.DefaultHardPodAffinityWeight {
-		args := kubeschedulerv1alpha2.InterPodAffinityArgs{
+		args := &kubeschedulerv1alpha2.InterPodAffinityArgs{
 			HardPodAffinityWeight: &o.HardPodAffinitySymmetricWeight,
 		}
-		profile.PluginConfig = append(profile.PluginConfig, plugins.NewPluginConfig(interpodaffinity.Name, args))
+		profile.PluginConfig = append(profile.PluginConfig, plugins.NewRawPluginConfig(interpodaffinity.Name, args))
 	}
 	return nil
 }
diff --git a/pkg/scheduler/framework/plugins/legacy_registry.go b/pkg/scheduler/framework/plugins/legacy_registry.go
index 2be7534c549..c94e48b31b9 100644
--- a/pkg/scheduler/framework/plugins/legacy_registry.go
+++ b/pkg/scheduler/framework/plugins/legacy_registry.go
@@ -501,6 +501,13 @@ func NewPluginConfig(pluginName string, args interface{}) config.PluginConfig {
 	}
 }
 
+func NewRawPluginConfig(pluginName string, obj runtime.Object) config.PluginConfig {
+	return config.PluginConfig{
+		Name: pluginName,
+		Args: runtime.RawExtension{Object: obj},
+	}
+}
+
 // ProcessPredicatePolicy given a PredicatePolicy, return the plugin name implementing the predicate and update
 // the ConfigProducerArgs if necessary.
 func (lr *LegacyRegistry) ProcessPredicatePolicy(policy config.PredicatePolicy, pluginArgs *ConfigProducerArgs) string {

```

Reply:
As the plan goes, we will eventually just use the internal types with no more serialization/deserialization required.

Reply:
SG.